### PR TITLE
Add doctr support, setup-guide

### PR DESCRIPTION
**Pull request by n-wbrown**
## Description
This PR adds doctr support to our cookiecutter module. Some maintenance has been done for compatibility with newer versions of sphinx.

Additionally creation hooks have been added to help guide users through the process of configuring doctr and versioneer.

## Motivation and Context
These features are intended to help developers instantly get started with developing new packages while following good coding conventions. The primary feature of this PR, the doctr compatibility, should facilitate good documentation practices and help make documentation more easily shared. Configuring doctr in the past has been a time consuming task even for experienced users that is prone to detail-based errors. These changes should prevent this time cost from being inccurred again for each new package. 

In the past, I've experienced some trouble remembering the individual steps to complete the additional configuration required for getting started with this cookiecutter. The hooks that now automatically complete these steps help us attain the ideal overhead-free cookiecutter. 

## How Has This Been Tested?
I've tested this by generating the [test_proj]( https://github.com/n-wbrown/test_proj) repository and ensuring that the documentation generation is functional.

## Where Has This Been Documented?
Notes have been added to the readme to with references on doctr and versioneer. The setup steps now include prompts for users to automatically or manually configure these features.

## Additional Note
I like having tools that help us follow best practices, save development time, and maintain consistent program structures. Additionally I think having tools that assist with coding style, project structure and documentation may be very useful for helping new interns get started with their own projects. Given the number of interns we expect in PCDS, it may be much more efficient to have one very well maintained cookiecutter as opposed to helping them start their own projects one-by-one.

### CODE REVIEW

File: run_tests.py [hunk 1]
```diff
@@ -56,13 +56,16 @@
         args.remove('--no-template')
         args.append('--ignore=tests')
         
+    # Set defautls for the test
```
Comment (ZLLentz) [inline run_tests.py]: typo here

File: {{ cookiecutter.folder_name }}/.travis.yml [hunk 2]
```diff
@@ -1,7 +1,17 @@
 language: python
+udo: false
```
Comment (ZLLentz) [inline {{ cookiecutter.folder_name }}/.travis.yml]: typo here, should be sudo not udo

File: {{ cookiecutter.folder_name }}/docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import sys
+import sphinx_rtd_theme
+module_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),'../../')
+sys.path.insert(0,module_path)
+
+# -- Project information -----------------------------------------------------
+
+project = '{{ cookiecutter.project_name }}'
+copyright = '2018, {{ cookiecutter.author_name }}'
+author = '{{ cookiecutter.author_name }}'
```
Comment (teddyrendahl) [inline {{ cookiecutter.folder_name }}/docs/source/conf.py]: I think this should just be `SLAC National Accelerator Laboratory`

File: run_tests.py (hunk 1 above)
Comment (n-wbrown) [reply run_tests.py]: I now realize this comment refers to deleted code. Removing

File: {{ cookiecutter.folder_name }}/docs/source/conf.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+import os
+import sys
+import sphinx_rtd_theme
+module_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),'../../')
+sys.path.insert(0,module_path)
+
+# -- Project information -----------------------------------------------------
+
+project = '{{ cookiecutter.project_name }}'
+copyright = '2018, {{ cookiecutter.author_name }}'
```
Comment (teddyrendahl) [inline {{ cookiecutter.folder_name }}/docs/source/conf.py]: We can auto-update the year! https://github.com/pcdshub/nabs/blob/master/docs/source/conf.py#L26

File: {{ cookiecutter.folder_name }}/.travis.yml [hunk 5]
```diff
@@ -17,18 +27,31 @@ install:
   # Useful for debugging any issues with conda
   - conda info -a
   #Grab all dependencies
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION coverage pip wheel pytest
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION pip wheel pytest
```
Comment (ZLLentz) [inline {{ cookiecutter.folder_name }}/.travis.yml]: consider using `dev_requirements.txt` here like in `hutch-python`

File: {{ cookiecutter.folder_name }}/.travis.yml [hunk 6]
```diff
@@ -17,18 +27,31 @@ install:
...
   #Launch Conda environment
   - source activate test-environment
   #Install pedl
```
Comment (ZLLentz) [inline {{ cookiecutter.folder_name }}/.travis.yml]: Somebody must have copy/pasted this comment a long time ago

File: {{ cookiecutter.folder_name }}/.travis.yml [hunk 7]
```diff
@@ -17,18 +27,31 @@ install:
...
   - pip install codecov
   - pip install -r requirements.txt
   #Install
-  - python setup.py install
+  - pip install -e .
```
Comment (ZLLentz) [inline {{ cookiecutter.folder_name }}/.travis.yml]: I'm a fan of the whole `conda build` stuff for the `hutch-python` tests, but I could see why you might not want that as the template

File: requirements.txt [hunk 8]
```diff
@@ -1,5 +1,7 @@
 cookiecutter
-pytest
+doctr
 simplejson
```
Comment (teddyrendahl) [inline requirements.txt]: Remove this?

File: {{ cookiecutter.folder_name }}/.travis.yml (hunk 5 above)
Comment (teddyrendahl) [reply {{ cookiecutter.folder_name }}/.travis.yml]: https://github.com/pcdshub/hutch-python/blob/695c6b75e1b8ecda4a32bccb66d3311363fa6532/.travis.yml#L37

File: {{ cookiecutter.folder_name }}/docs/source/conf.py [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+    'sphinx.ext.todo',
+    'sphinx.ext.coverage',
+    'sphinx.ext.mathjax',
+    'sphinx.ext.viewcode',
+    'sphinx.ext.githubpages',
+]
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
```
Comment (ZLLentz) [inline {{ cookiecutter.folder_name }}/docs/source/conf.py]: putting `autosummary_generate = True` somewhere in this file lets us use the sphinx autosummary features in our docs

File: {{ cookiecutter.folder_name }}/docs/source/conf.py (hunk 4 above)
Comment (ZLLentz) [inline {{ cookiecutter.folder_name }}/docs/source/conf.py]: This shouldn't be hard-coded at 2018

File: {{ cookiecutter.folder_name }}/docs/source/conf.py (hunk 3 above)
Comment (n-wbrown) [reply {{ cookiecutter.folder_name }}/docs/source/conf.py]: That's the default value for `cookiecutter.author_name`.

**Pull request merged**